pkg/utils/utils.go

```diff
@@ -372,6 +377,86 @@ func GetNetNames(pciAddr string) ([]string, error) {
 	return names, nil
 }
 
+// GetNetNamesFromNetns returns host net interface names as string for a PCI device from its pci address,
+// given that the host net interface has been moved into a child network namespace
+func GetNetNamesFromNetns(pciAddr string) ([]string, error) {
+	networkNamespaces, err := GetNetworkNamespaces()
+	if err != nil {
+		return nil, fmt.Errorf("GetNetNamesFromNetns(): failed to get network namespaces: %q", err)
+	}
+
+	// this temporary directory will be used as a mount point for each child namespace
+	tmpDir, err := os.MkdirTemp("/tmp", "")
+	if err != nil {
+		return nil, fmt.Errorf("GetNetNamesFromNetns(): could not create temp dir for mounting namespaces")
+	}
+
+	names := make([]string, 0)
+	for _, networkNamespace := range networkNamespaces {
+		netNsDir := filepath.Join(varRunNetns, networkNamespace)
+		_ = ns.WithNetNSPath(netNsDir, func(childNs ns.NetNS) error {
+			// mount the network namespace to the previously defined temp directory
+			if err := syscall.Mount(netNsDir, tmpDir, "sysfs", 0, ""); err != nil {
+				return fmt.Errorf("GetNetNamesFromNetns(): failed to mount /sys: %v", err)
+			}
+
+			defer func() {
+				_ = syscall.Unmount(tmpDir, 0)
+			}()
+
+			// search for the interface
+			interfaceDir := fmt.Sprintf("%v/bus/pci/devices/%v/net", tmpDir, pciAddr)
+			fInfos, err := os.ReadDir(interfaceDir)
+			if err != nil {
+				return fmt.Errorf("GetNetNamesFromNetns(): unable to read %v: %w", interfaceDir, err)
+			}
+
+			for _, fInfo := range fInfos {
+				ifaceName := fInfo.Name()
+				alias, _ := GetInterfaceAlias(ifaceName)
```

Comment:
would this contain the original interface name ?

if statement below indicates that this might not be set for netdev ?

i need to give this one a try on a system to be able to provide proper feedback.

with this, we could potentially determine the link type as well when creating the device in `gen_net.go`

we have a community meeting tommorow so i will add this PR to the agenda to get some attention to this PR and have others provide feedback as well.